lib/index.js

```diff
@@ -46,42 +46,65 @@ $(document).ready(() => {
       var mealName = $(event.target).attr('data').split('-')[0]
-      foodsDiary.deleteFoodFromMeal(mealName, foodId)
+      foodsDiary.updateMealWithFood(mealName, foodId, 'DELETE')
       removeFoodRow()
+      updateMealCalories(setFoodData(event.target), mealName, 'decreaseMeal')
     }
   })
 })
 
-const updateMealCalories = (food, meal) => {
-  updateMealTable(food, meal)
-  updateTotalsTable(food)
+const updateMealCalories = (food, meal, method) => {
+  updateMealTable(food, meal, method)
+  updateTotalsTable(food, method)
 }
 
-const updateMealTable = (food, meal) => {
+const updateMealTable = (food, meal, method) => {
   var currentCaloriesNode = $(`#${meal}-total-calories-count`).find('p')
   var totalCaloriesNode = getRemainingCaloriesValue(meal)
-  updateRemainingCaloriesValue(food, meal, totalCaloriesNode)
-  updateTotalCaloriesValue(food, currentCaloriesNode)
+  updateTotalCaloriesValue(food, currentCaloriesNode, method)
+  updateRemainingCaloriesValue(food, meal, totalCaloriesNode, method)
 }
 
-const updateTotalsTable = (food) => {
+const updateTotalsTable = (food, method) => {
   var totalDiaryConsumedNode = $('.diary-total-calories-consumed')
   var currentDiaryRemainingCaloriesNode = getRemainingCaloriesValue('diary')
-  updateTotalCaloriesValue(food, totalDiaryConsumedNode)
-  updateRemainingCaloriesValue(food, 'diary', currentDiaryRemainingCaloriesNode)
+  updateTotalCaloriesValue(food, totalDiaryConsumedNode, method)
+  updateRemainingCaloriesValue(food, 'diary', currentDiaryRemainingCaloriesNode, method)
 }
 
-const updateRemainingCaloriesValue = (food, meal, currentCaloriesNode) => {
-  var updatedRemainingCalories = currentCaloriesNode.text() - food.calories
+const updateRemainingCaloriesValue = (food, meal, currentCaloriesNode, method) => {
+  if (method == 'decreaseMeal') {
+    var operator = '+'
+  } else {
+    var operator = '-'
+  }
+  var totalCalories = parseInt(currentCaloriesNode.text())
+  var foodCalories = parseInt(food.calories)
+  var updatedRemainingCalories = calculateTotals(operator, totalCalories, foodCalories)
   updateNodeValue(currentCaloriesNode, updatedRemainingCalories)
   foodsDiary.styleRemainingCalorieCount(meal, updatedRemainingCalories)
 }
 
-const updateTotalCaloriesValue = (food, totalCaloriesNode) => {
-  var updatedTotalCalories = parseInt($(totalCaloriesNode).text()) + parseInt(food.calories)
+const updateTotalCaloriesValue = (food, totalCaloriesNode, method) => {
+  if (method == 'decreaseMeal') {
+    var operator = '-'
+  } else {
+    var operator = '+'
+  }
+  var totalCalories = parseInt($(totalCaloriesNode).text())
+  var foodCalories = parseInt(food.calories)
+  var updatedTotalCalories = calculateTotals(operator, totalCalories, foodCalories)
   updateNodeValue(totalCaloriesNode, updatedTotalCalories)
 }
 
+const calculateTotals = (operator, digit1, digit2) => {
+  var formulate = {
+    '+': function (digit1, digit2) { return digit1 + digit2 },
```

Comment:
This function declaration is in ES5 syntax.

Will you change it to ES6? I pulled down the branch and found that this is a way to use ES6 syntax and keep functionality.
```
'+' : (digit1, digit2) => { return digit1 + digit2 },
'-' : (digit1, digit2) => { return digit1 - digit2 }
```

Here's a link to the documentation about it: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Method_definitions

Reply:
Good catch! Syntax has been updated to ES6.